lib/passport-wsfed-saml2/samlp.js

```diff
@@ -213,7 +226,26 @@ Samlp.prototype = {
   },
 
   decodeResponse: function(req) {
-    var decoded = new Buffer(req.body['SAMLResponse'], 'base64').toString();
+    var decoded = new Buffer(req.body['SAMLResponse'], 'base64').toString(this.options.encoding);
+
+    if (!this.options.explicit_encoding){
```

Comment:
if you already wrote the code to extract the prolog encoding, then there's no need to add encoding to options.
I'd go for one or the other: select your IdP encoding by options or let the module parse and decide which one to use.
Maybe for now it's simpler to trust the option (defaulting to utf8), later we can work on a xml-prolog-parser, so if the encoding is not provided we would try to extract it from xml, defaulting to utf8 if it fails.
Wdyt?

Reply:
what if the SAML response is not sending an encoding ? an option would help users decide it. If we go with the option alternative we will miss those cases where the IdP con send one or the other.

I think that we need both cases. My main doubt is which of them should prevail. If you have both, which one should we use.

Reply:
TODO

- Extract logic to a separate function that understands XML and returns the encoding
- Change encoding to default encoding
- Remove explicit encoding because encoding from XML should prevail from the option
- Wrap the function in a try/catch to prevent issues.

Reply:
@mcastany Does this reflect a change from the initial approach we discussed?
> Remove explicit encoding because encoding from XML should prevail from the option